# task_manager_params.cfg
#!/usr/bin/env python

# 1.导包
from dynamic_reconfigure.parameter_generator_catkin import *
PACKAGE = "task_manager"

# 2.创建生成器
gen = ParameterGenerator()

# 3.向生成器添加若干参数
#add(name, paramtype, level, description, default=None, min=None, max=None, edit_method="")
gen.add("landmark_thred", double_t, 0, "过点阈值", 0.0, -1.0, 1.0)
gen.add("derail_limit_x", double_t, 0, "x轴脱轨阈值", 0.0, -1.0, 1.0)
gen.add("derail_limit_y", double_t, 0, "y轴脱轨阈值", 0.0, -1.0, 1.0)
gen.add("derail_limit_theta", double_t, 0, "角度脱轨阈值", 0.0, -3.14, 3.14)
gen.add("static_work_thred", double_t, 0, "举升阈值", 0.2, -1.0, 1.0)                                                                         

# 4.生成中间文件并退出
exit(gen.generate(PACKAGE,"task_manager","task_manager_params")) #注意最后一个参数必须是cfg文件名
